project/core/sourcecontrol/PvcsHistoryParser.cs

```diff
@@ -12,9 +12,11 @@ namespace ThoughtWorks.CruiseControl.Core.Sourcecontrol
     /// </summary>
 	public class PvcsHistoryParser : IHistoryParser
 	{
-		private static Regex _searchRegEx = new Regex(@"(?<Archive>Archive:\s+.*?\r\n(.|\s)*?(={35}(\r\n|$)))", RegexOptions.Compiled | RegexOptions.IgnorePatternWhitespace | RegexOptions.ExplicitCapture | RegexOptions.IgnoreCase);
-		private static Regex _archiveRegEx = new Regex(@"Archive:\s+(?<ArchiveName>.*?)\nWorkfile:\s+(?<Filename>.*?)\nArchive\screated:\s+(?<CreatedDate>.*?)\r\n(.|\s)*?-{35}\r\n(?<Revision>Rev\s\d+(\.\d+)*\r\n(.*\r\n)*?Author\sid:.*?\r\n((?!(={35}|-{35}))(.|\s)*?\r\n)?(-{35}|={35})(\r\n|$))+", RegexOptions.Compiled | RegexOptions.IgnorePatternWhitespace | RegexOptions.ExplicitCapture | RegexOptions.IgnoreCase);
-		private static Regex _revisionRegEx = new Regex(@"Rev\s(?<Version>\d(\.\d+)*)\r\n(.*?\r\n)?Checked\sin:\s+(?<CheckIn>.*?)\r\n(.*?\r\n)?Last\smodified:\s+(?<PreviousModification>.*?)\r\n(.*?\r\n)?Author\sid:\s+(?<Author>.*?)\s.*?\r\n(Branches:\s+.*?\r\n)?(?<Comment>(((?!(={35}|-{35}))(.|\s)*?)\r\n)?)(={35}|-{35})(\r\n|$)", RegexOptions.Compiled | RegexOptions.IgnorePatternWhitespace | RegexOptions.ExplicitCapture | RegexOptions.IgnoreCase);
+        private static string RegexNewLine = Regex.Escape(Environment.NewLine); 
+    
+		private static Regex _searchRegEx = new Regex(@"(?<Archive>Archive:\s+.*?" + RegexNewLine + @"(.|\s)*?(={35}(" + RegexNewLine + @"|$)))", RegexOptions.Compiled | RegexOptions.IgnorePatternWhitespace | RegexOptions.ExplicitCapture | RegexOptions.IgnoreCase);
```

Comment:
You can also use $@"bla bla {variable}"

Reply:
Ah well, I'm a bit old school here, I didn't know about this combination here. Do you want me to change the code?

Reply:
This should work
```
$@"(?<Archive>Archive:\s+.*?{RegexNewLine}(.|\s)*?(={{35}}({RegexNewLine}|$)))";
```
Curly braces from ```{35}``` are doubled (escaped) so that the curly-braces block will not be interpreted as variable 35 in this context.
I think it's more readable this way but I might be too used to this sugar syntax.

Reply:
Well, I find it less readable for the `_revisionRegEx` static that has lots of carriage return inside it.
To me, the most readable way would be something like this:
```
        private static Regex _revisionRegEx = 
            new Regex(
                @"Rev\s(?<Version>\d(\.\d+)*)" + RegexNewLine + 
                @"(.*?" + RegexNewLine + @")?" + 
                @"Checked\sin:\s+(?<CheckIn>.*?)" + RegexNewLine + 
                @"(.*?" + RegexNewLine + @")?" +
                @"Last\smodified:\s+(?<PreviousModification>.*?)" + RegexNewLine + 
                @"(.*?" + RegexNewLine + @")?" +
                @"Author\sid:\s+(?<Author>.*?)\s.*?" + RegexNewLine + 
                @"(Branches:\s+.*?" + RegexNewLine + @")?" +
                @"(?<Comment>(((?!(={35}|-{35}))" + @"(.|\s)*?)" + RegexNewLine + @")?)" +
                @"(={35}|-{35})" + @"(" + RegexNewLine + @"|$)", 
                RegexOptions.Compiled | RegexOptions.IgnorePatternWhitespace | RegexOptions.ExplicitCapture | RegexOptions.IgnoreCase
            );
```
This allows having the lines all by themselves along with keeping the regex groups on the same line as well. 
I mean, what's hard here, is to understand the regex and its groups.

But mixing the `$@` syntax in there should be fine as well, like this:
```
        private static Regex _revisionRegEx = 
            new Regex(
                $@"Rev\s(?<Version>\d(\.\d+)*){RegexNewLine}" + 
                $@"(.*?{RegexNewLine})?" + 
                $@"Checked\sin:\s+(?<CheckIn>.*?){RegexNewLine}" + 
                $@"(.*?{RegexNewLine})?" +
                $@"Last\smodified:\s+(?<PreviousModification>.*?){RegexNewLine}" + 
                $@"(.*?{RegexNewLine})?" +
                $@"Author\sid:\s+(?<Author>.*?)\s.*?{RegexNewLine}" + 
                $@"(Branches:\s+.*?{RegexNewLine})?" +
                @"(?<Comment>(((?!(={35}|-{35}))" + $@"(.|\s)*?){RegexNewLine})?)" +
                @"(={35}|-{35})" + $@"({RegexNewLine}|$)", 
                RegexOptions.Compiled | RegexOptions.IgnorePatternWhitespace | RegexOptions.ExplicitCapture | RegexOptions.IgnoreCase
            );
```
What do you think?

Reply:
I think your first form of _revisionRegEx is quite readable.